validator/src/journal/journal_ffi.rs

```diff
@@ -68,11 +73,30 @@ use py_object_wrapper::PyObjectWrapper;
 struct Journal {
     chain_controller: ChainController,
     executor: Option<Executor>,
+    genesis_controller: GenesisController,
 }
 
 impl Journal {
-    fn start(&mut self) {
+    fn start(&mut self) -> ErrorCode {
+        match self.genesis_controller.requires_genesis() {
+            Ok(create_genesis) => {
+                if create_genesis {
+                    if let Err(err) = self.genesis_controller.start() {
+                        error!("Unable to create genesis block: {}", err);
+                        return ErrorCode::Unknown;
```

Comment:
These should be genesis specific errors. When this fails, it logs badly.

Reply:
That was updated in a later commit